core/src/main/scala/org/apache/spark/sql/execution/columnar/JDBCSourceAsStore.scala

```diff
@@ -184,20 +185,21 @@ final class ColumnBatchIteratorOnRS(conn: Connection,
     fetchColQuery: String)
     extends ResultSetIterator[Array[Byte]](conn, stmt, rs, context) {
   var currentUUID: String = _
-  val ps = conn.prepareStatement(fetchColQuery)
-  var colBuffers: Option[scala.collection.mutable.HashMap[Int, (ByteBuffer, Blob)]] = None
+  val ps: PreparedStatement = conn.prepareStatement(fetchColQuery)
+  var colBuffers: Option[TIntObjectHashMap] = None
+
   def getColumnLob(bufferPosition: Int): ByteBuffer = {
     colBuffers match {
-      case Some(map) => map(bufferPosition)._1
+      case Some(map) =>
+        map.get(bufferPosition).asInstanceOf[(ByteBuffer, Blob)]._1
```

Comment:
Agree that TIntObjectHashMap is not nice to use. As part of SNAP-1454 have added fastutil reference for use elsewhere (which is type safe) and we should make a point to use that every where. Can also gradually transition trove maps to it and take out trove like done in Geode.

The main reason for using the open hash maps is much lesser garbage more than performance (though fastutil maps are also quite a bit faster than JDK/Scala maps). Though it could be true that garbage here may not be a contributor but to be on the safe side we should just make that a standard practice and lessen our overall GC troubles.

Will change this to use fastutil and add the reference in this PR itself.